core/src/main/scala/kafka/coordinator/group/GroupMetadata.scala

```diff
@@ -236,12 +244,47 @@ private[group] class GroupMetadata(val groupId: String, initialState: GroupState
       leaderId = members.keys.headOption
   }
 
+  /**
+    * [For static members only]: Replace the old member id with the new one,
+    * keep everything else unchanged and return the updated member.
+    */
+  def replace(oldMemberId: String,
+              newMemberId: String,
+              groupInstanceId: Option[String]): MemberMetadata = {
```

Comment:
I think adding an assertion here would be helpful. Would be messy if we do null check in caller every time when we call `replace`, `addStaticMember` and `getStaticMemberId`, what do you think?

Reply:
How about rename it to `maybeReplaceGroupInstance` and do the check in the callee and make it no-op if `groupInstanceId` is empty then?

Reply:
I agree with renaming but feel slightly against second proposal. I think the goal is to avoid people from passing in null group.instance.id in the code level. @guozhangwang